deeppavlov/models/multitask_pal_bert/multitask_pal_bert.py

```diff
@@ -0,0 +1,468 @@
+                    "task for inference")
+            return {
+                task_names[0]: list(kwargs.values()) if kwargs else list(args)}
+
+        if all([isinstance(task_distr, int)
+               for task_distr in distribution.values()]):
+            ints = True
+        elif all(
+            [isinstance(task_distr, list)
+             for task_distr in distribution.values()]
+        ):
+            ints = False
+        else:
+            raise ConfigError(
+                f"Values of `{what_to_distribute}_distribution` attribute of `MultiTaskBert` have to be "
+                f"either `int` or `list` not both. "
+                f"{what_to_distribute}_distribution = {distribution}")
+
+        args_by_task = {}
+
+        flattened = []
+        for task_name in task_names:
+            if isinstance(task_name, str):
+                flattened.append(task_name)
+            else:
+                flattened.extend(task_name)
+        task_names = flattened
+
+        if args and not ints:
+            ints = True
+            distribution = {
+                task_name: len(in_distr) for task_name,
+                in_distr in distribution.items()}
+        if ints:
+            if kwargs:
+                values = list(kwargs.values())
+            else:
+                values = args
+            n_distributed = sum([n_args for n_args in distribution.values()])
+            if len(values) != n_distributed:
+                raise ConfigError(
+                    f"The number of '{what_to_distribute}' arguments of MultitaskBert does not match "
+                    f"the number of distributed params according to '{what_to_distribute}_distribution' parameter. "
+                    f"{len(values)} parameters are in '{what_to_distribute}' and {n_distributed} parameters are "
+                    f"required '{what_to_distribute}_distribution'. "
+                    f"{what_to_distribute}_distribution = {distribution}")
+            values_taken = 0
+            for task_name in task_names:
+                args_by_task[task_name] = {}
+                n_args = distribution[task_name]
+                args_by_task[task_name] = [values[i]
+                                           for i in range(values_taken, values_taken + n_args)]
+                values_taken += n_args
+
+        else:
+            assert kwargs
+            arg_names_used = []
+            for task_name in task_names:
+                in_distr = distribution[task_name]
+                args_by_task[task_name] = {}
+                args_by_task[task_name] = [kwargs[arg_name]
+                                           for arg_name in in_distr]
+                arg_names_used += in_distr
+            set_used = set(arg_names_used)
+            set_all = set(kwargs.keys())
+            if set_used != set_all:
+                raise ConfigError(
+                    f"There are unused '{what_to_distribute}' parameters {set_all - set_used}"
```

Comment:
What if some value exists in set_used but not in set_all?

Reply:
Because all the number of parameters need to be used, for example: `{"task1": 2, "task2": 3}` meaning 2 parameters for task1 and 3 for task2, so all of them should be used. If in some cases some parameter is unused that is an error.

Reply:
Yes, it is, but if some value exists in set_used but not in set_all error message will be uncomprehensive, as set_all - set_used is empty. Are you sure that it will never happen?

Reply:
Added a condition for the case when `len(set_used) > len(set_all)`